avm/res/cache/redis-enterprise/main.bicep

```diff
@@ -0,0 +1,490 @@
+// Resources      //
+// ============== //
+
+#disable-next-line no-deployments-resources
+resource avmTelemetry 'Microsoft.Resources/deployments@2024-03-01' = if (enableTelemetry) {
+  name: '46d3xbcp.res.cache-redisenterprise.${replace('-..--..-', '.', '-')}.${substring(uniqueString(deployment().name, location), 0, 4)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+      outputs: {
+        telemetry: {
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
+
+resource cMKKeyVault 'Microsoft.KeyVault/vaults@2023-02-01' existing = if (!empty(customerManagedKey.?keyVaultResourceId)) {
+  name: last(split((customerManagedKey.?keyVaultResourceId ?? 'dummyVault'), '/'))
+  scope: resourceGroup(
+    split((customerManagedKey.?keyVaultResourceId ?? '//'), '/')[2],
+    split((customerManagedKey.?keyVaultResourceId ?? '////'), '/')[4]
+  )
+
+  resource cMKKey 'keys@2023-02-01' existing = if (!empty(customerManagedKey.?keyVaultResourceId) && !empty(customerManagedKey.?keyName)) {
+    name: customerManagedKey.?keyName ?? 'dummyKey'
+  }
+}
+
+resource cMKUserAssignedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2023-01-31' existing = if (!empty(customerManagedKey.?userAssignedIdentityResourceId)) {
+  name: last(split(customerManagedKey.?userAssignedIdentityResourceId ?? 'dummyMsi', '/'))
+  scope: resourceGroup(
+    split((customerManagedKey.?userAssignedIdentityResourceId ?? '//'), '/')[2],
+    split((customerManagedKey.?userAssignedIdentityResourceId ?? '////'), '/')[4]
+  )
+}
+
+resource redisCluster 'Microsoft.Cache/redisEnterprise@2024-09-01-preview' = {
+  name: name
+  location: location
+  tags: tags
+  identity: identity
+  properties: {
+    encryption: !empty(customerManagedKey)
+      ? {
+          customerManagedKeyEncryption: {
+            keyEncryptionKeyIdentity: !empty(customerManagedKey.?userAssignedIdentityResourceId)
+              ? {
+                  identityType: 'userAssignedIdentity'
+                  userAssignedIdentityResourceId: cMKUserAssignedIdentity.id
+                }
+              : null
+            keyEncryptionKeyUrl: !empty(customerManagedKey.?keyVersion ?? '')
+              ? '${cMKKeyVault::cMKKey.properties.keyUri}/${customerManagedKey!.keyVersion}'
+              : cMKKeyVault::cMKKey.properties.keyUriWithVersion
+          }
+        }
+      : null
+    highAvailability: isAmr ? highAvailability : null
+    minimumTlsVersion: minimumTlsVersion
+  }
+  sku: {
+    capacity: isEnterprise ? capacity : null
+    name: skuName
+  }
+  zones: !empty(availabilityZones) ? availabilityZones : null
+}
+
+module redisCluster_database 'database/main.bicep' = {
+  name: '${uniqueString(deployment().name, location)}-redis-database'
```

Comment:
I just gotta ask - can a Redis-Enterprise instance only have a single database?

Reply:
Yes correct

Reply:
Interesting. Thanks for clarifying sir